### ISSUE TITLE
[de][LibreOffice] false positive bei "Donnerhall" und "all"

### ISSUE THREAD

**Issue by ChristianPervoelz** [languagetool-org/languagetool#5530]
**Beispielsatz**
Der Donnerhall verklang nur langsam in meinen Ohren. Das war mir in all den Jahren meines Lebens noch nicht passiert.

**Problem**
Die Wörter _Donnerhall_ und _all_ werden als "im nachfolgenden/vorgehenden Satz verwendet" markiert.

**Info**
LibreOffice 7.0.4.2
LT Plugin: 5.4
Alle Stilregeln aktiviert

**Comment by danielnaber** [comment]
@FredKruse This seems to be a general issue. Can we fix it without adding special cases?

**Comment by FredKruse** [comment]
I excluded the word "all".
A general solution could be to exclude the function to test compound words, but that would be a big limit of functionality.

**Comment by ChristianPervoelz** [comment]
Without any knowledge about the code and the feasibility, I'd like to give a proposal for a more general solution. It is a partial solution only, but for most of the issue I raised, it might work pretty well.

1. Check if one of the words is a compound word and the other is **not**
2. Check that the non-compound is **not equal** (in the sense of equality of LT) to any of the words the compound word is composed of

I did a check with quick&dirty implementation and it looked quite promising to me.
